kafka-client/src/main/java/dev/responsive/kafka/store/CommitBuffer.java

```diff
@@ -74,6 +83,20 @@ class CommitBuffer<K> implements RecordBatchingStateRestoreCallback {
 
     this.buffer = new TreeMap<>(plugin);
     this.truncateChangelog = truncateChangelog;
+    this.subPartitioner = subPartitioner;
+  }
+
+  public void init() {
+    subPartitioner.all(partition).forEach(p -> client.initializeOffset(tableName, p));
+
+    // when we initialize a new state store, we should revoke all ongoing
+    // transactions to the remote store - otherwise it is possible that
+    // there's a hanging transaction that will fence this writer if there
+    // is no data to restore
+    final long offset = offset();
+    if (offset >= 0) {
+      subPartitioner.all(partition).forEach(p -> client.revokePermit(tableName, p, offset));
```

Comment:
This will only revoke ongoing transactions for the sub-partitions at the lowest offset. I _think_ that's fine because we only ever acquire a permit if we are writing records, so the transaction will be cleared when the restore hits the record that was being written. But I'm having trouble convincing myself. Need to think about it a little more.

Reply:
Yes that's correct - it's ok because that's also the offset that we restore from, so we will eventually restore an offset that is >= other open transactions, at which point they will also be revoked. 

We ignore fenced exceptions when restoring, we just move on to the next batch.

Reply:
Yeah I am convinced now. One more thing - don't you need to execute these? The client call just creates a statement right?

Reply:
🤦 let me add a test for this... I forgot that `initializeOffset` executes it but `revokePermit` does not.

Reply:
Fixed - also added `@CheckReturnValue` to the statements in `CassandraClient` that only return a `BoundStatement` and don't execute anything so doing this will result in a warning going forward.